### PR TITLE
Better test coverage

### PR DESCRIPTION
**Pull request by ciroque**
### Proposed changes

Add more tests. Non-official goal of 85% test coverage on each and every module.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [X] I have read the [`CONTRIBUTING`](https://github.com/nginxinc/nginx-loadbalancer-kubernetes/blob/main/CONTRIBUTING.md) document
- [X] If applicable, I have added tests that prove my fix is effective or that my feature works
- [X] If applicable, I have checked that any relevant tests pass after adding my changes
- [X] I have updated any relevant documentation ([`README.md`](https://github.com/nginxinc/nginx-loadbalancer-kubernetes/blob/main/README.md) and [`CHANGELOG.md`](https://github.com/nginxinc/nginx-loadbalancer-kubernetes/blob/main/CHANGELOG.md))

### CODE REVIEW

File: internal/probation/server_test.go [hunk 1]
```diff
@@ -51,3 +53,21 @@ func TestHealthServer_HandleFailCheck(t *testing.T) {
...
+	response, err := http.Get("http://localhost:51031/livez")
+	if err != nil {
+		t.Error(err)
+	}
+
+	if response.StatusCode != http.StatusOK {
+		t.Errorf("Expected status code %v, got %v", http.StatusAccepted, response.StatusCode)
+	}
+
+	logrus.Infof("recevied a response from the probe server: %v", response)
```
Comment (4141done) [inline internal/probation/server_test.go:70]: nit: received

File: internal/probation/server_test.go [hunk 2]
```diff
@@ -51,3 +53,21 @@ func TestHealthServer_HandleFailCheck(t *testing.T) {
...
+
+	server.Stop()
```
Comment (4141done) [inline internal/probation/server_test.go:72]: Not experienced with go testing but do we need to make sure this server is stopped if the test terminates before this point?

**Review by 4141done — APPROVED** [review]
👍 🐳 👍 
Looks good! minor nits and questions but nothing blocking

File: internal/probation/server_test.go (hunk 1 above)
Comment (ciroque) [reply internal/probation/server_test.go:70]: just noticed that too

File: internal/probation/server_test.go (hunk 2 above)
Comment (ciroque) [reply internal/probation/server_test.go:72]: it should get cleaned up, but I'll put it in a `defer` to be sure

**Pull request merged**